datanode/gateway/graphql/schema.graphql

```diff
@@ -1943,6 +1943,14 @@ type Market {
 
   "Optional: Market ID of the successor to this market if one exists"
   successorMarketID: ID
+
+  "Funding Periods for perpetual markets"
+  fundingPeriods(
```

Comment:
**question (blocking)**: Checking other platforms, it looks like funding periods typically take a date parameter or two:

(dydx v4 uses `.../v4/historicalFunding/BTC-USD?effectiveBeforeOrAt=2023-07-06T21:59:59.899Z&null=2023-07-02T18:59:59.899Z`, okx `.../funding-rate-history-new?t=1690480575735&ccy=BTC`) . Is there a reason we don't provide that? 

Presumably this would be used for charting the funding rates. Date controls would help there.

**Basic suggestion**: An optional new parameter, `endDate`, that can be used to fetch `n` funding periods up until a dateTime (where `n` is defined by pagination or a sensible default)

**Fancier suggestion**: Two optional parameters, `startDate` and `endDate`.

Is there arny reason we can't get those?

Reply:
There's no reason we can't add it, happy to do so.